streams/src/test/java/org/apache/kafka/streams/KafkaStreamsTest.java

```diff
@@ -214,6 +214,70 @@ public void testStateOneThreadDeadButRebalanceFinish() throws InterruptedExcepti
         Assert.assertEquals(KafkaStreams.State.NOT_RUNNING, globalStreams.state());
     }
 
+    @Test
+    public void testStateOneAllThreadDead() throws InterruptedException {
+        final StateListenerStub stateListener = new StateListenerStub();
+        globalStreams.setStateListener(stateListener);
+
+        Assert.assertEquals(0, stateListener.numChanges);
+        Assert.assertEquals(KafkaStreams.State.CREATED, globalStreams.state());
+
+        globalStreams.start();
+
+        TestUtils.waitForCondition(
+            () -> stateListener.numChanges == 2,
+            "Streams never started.");
+        Assert.assertEquals(KafkaStreams.State.RUNNING, globalStreams.state());
+
+        for (final StreamThread thread: globalStreams.threads) {
```

Comment:
Doesn't checkstyle require a space before `:` ?

Reply:
I think checkstyle does not enforce it, but I also like adding the space before `:` so I will do it :)